src/visualizers/VisualizerFactorHistogram.ts

```diff
@@ -0,0 +1,200 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from './VisualizerDefault'
+import {natlog} from '../shared/math'
+
+/** md
+# Factor Histogram
+
+(example image should go here)
+
+This visualizer takes the prime factorization of each element in the sequence
+and then creates a histogram of the number of elements of the sequence
+with X prime factors. the number of elements with X prime factors is along
+the x-axis. And the height is how many elements are in each bin of X.
+
+## Parameters
+**/
+
+class FactorHistogramVisualizer extends VisualizerDefault {
+    name = 'Histogram'
+
+    binSize = 1
+    terms = 100
+    firstIndex = this.seq.first
+
+    params = {
+        binSize: {
+            value: this.binSize,
+            forceType: 'integer',
+            displayName: 'Bin Size',
+            required: true,
+        },
+
+        firstIndex: {
+            value: this.firstIndex,
+            forceType: 'integer',
+            displayName: 'First Index',
+            required: true,
+        },
+
+        terms: {
+            value: this.terms,
+            forceType: 'integer',
+            displayName: 'How many terms of the series',
+            required: true,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Obtain the true first index
+    startIndex(): number {
+        return Math.max(this.firstIndex, this.seq.first)
+    }
+
+    // Obtain the true number of terms
+    numTerms(): number {
+        return Math.min(this.terms + this.startIndex(), this.seq.last)
+    }
+
+    // Find the largest value of the sequence asked for
+    largestValue(): bigint {
+        let largest_value = this.seq.getElement(this.startIndex())
+        for (let i = this.startIndex(); i < this.numTerms(); i++) {
+            const value = this.seq.getElement(i)
+            if (value > largest_value) {
+                largest_value = value
+            }
+        }
+        return largest_value
+    }
+
```

Comment:
You don't need the largest value of the sequence. You just need the largest value of the factorArray. Then you won't need to bother with the logs, and you won't need to search for the biggest nonempty bin.